COMPILING.md

```diff
@@ -0,0 +1,159 @@
+
+#### For all desktop distributions
+
+[QjackCtl](https://qjackctl.sourceforge.io/) is optional, but recommended to configure JACK.
+
+
+### Standard desktop build
+
+~~~
+qmake Jamulus.pro
+make clean
+make
+sudo make install
+~~~
+
+This puts the Jamulus binary into `/usr/local/bin`.
+
+#### Notes
+
+* The "make clean" command is essential to remove the automatically generated Qt files which are present in the .tar.gz file and may not match the Qt version you are using.
+
+* To use an external shared OPUS library instead of the built-in one use qmake `"CONFIG+=opus_shared_lib" Jamulus.pro`.
+
+* To use this file configure the software with `qmake "CONFIG+=noupcasename" Jamulus.pro` to make sure the output target name of this software is **j**amulus instead of **J**amulus.
+
+* Users of Raspberry Pi: You may want to compile the client on another machine and run the binary on the Raspberry Pi. In which case the only libraries you need to run it are those for a [headless server](Server-Linux#running-a-headless-server) build, but _with_ the JACK sound packages. In particular, have a look at the footnote for the headless build.
+
+* As of version 3.5.3, Jamulus is no longer compatible with Qt4.
+
+
+### “Headless” server build
+
+Note that you don’t need to install the JACK package(s) for a headless build. If you plan to run headless on Gentoo, or are compiling under Ubuntu for use on another Ubuntu machine, see the footnote.
+
+Also, although not strictly necessary, we recommend using the headless flag to speed up the build process. Gentoo users may also be able to avoid installing some dependencies as a consequence. For more information [see footnote](#footnote).
+
+Compile the sources to ignore the JACK sound library:
+
+~~~
+qmake "CONFIG+=nosound headless" Jamulus.pro
+make clean
+make
+~~~
+
+To control the server with systemd, see this [unit file example](https://github.com/jamulussoftware/jamulus/blob/master/distributions/jamulus-server.service). See also runtime [configuration options](/wiki/Command-Line-Options), and [this information](/wiki/Tips-Tricks-More#controlling-recording-on-linux-headless-servers) on controlling recordings on headless servers.
+
+### Notes
+
+* Compiling with the headless flag means you can avoid installing some of the dependent packages, save some disk space and/or speed up your build time under the following circumstances:
+
+* If you plan to run Jamulus on Gentoo Linux, the only packages you should need for a headless build are `qtcore`, `qtnetwork`, `qtconcurrent` and `qtxml` (both for building and running the server).
+
+* If you are running Jamulus on Ubuntu/Debian, you will need all dependent packages to compile the binary, but to run the resulting headless Jamulus server you should only need `libqt5core5a`, `libqt5network5`, `libqt5xml5` and probably `libqt5concurrent5`. This may be useful for compiling/upgrading on one machine to run the binary on another (a Raspberry Pi, for example).
```

Comment:
I'd put this above to explain why headless might be better for some people.

Reply:
Is it me, or is it very hard to understand what it being commented on in this Github thing (what is "this" in your comment)? I thought we agreed to remove the second para.

Reply:
By this I meant line 83. But never mind. Will do a new review now.